numpy/linalg/tests/test_linalg.py

```diff
@@ -1929,6 +1929,41 @@ def test_vector_as_first_and_last_argument(self):
         # the result should be a scalar
         assert_equal(multi_dot([A1d, B, C, D1d]).shape, ())
 
+    def test_basic_function_with_three_arguments_with_out_argument(self):
+        # multi_dot with three arguments uses a fast hand coded algorithm to
+        # determine the optimal order. Therefore test it separately.
+        A = np.random.random((6, 2))
+        B = np.random.random((2, 6))
+        C = np.random.random((6, 2))
+
+        out = np.zeros((6, 2))
+        ret = multi_dot([A, B, C], out=out)
+        assert out is ret
+        assert_almost_equal(out, A.dot(B).dot(C))
+        assert_almost_equal(out, np.dot(A, np.dot(B, C)))
+
+    def test_basic_function_with_two_arguments_with_out_argument(self):
+        # separate code path with two arguments
+        A = np.random.random((6, 2))
+        B = np.random.random((2, 6))
+        out = np.zeros((6, 6))
+        ret = multi_dot([A, B], out=out)
+        assert out is ret
+        assert_almost_equal(out, A.dot(B))
+        assert_almost_equal(out, np.dot(A, B))
+
+    def test_basic_function_with_dynamic_programing_optimization_with_out_argument(self):
+        # multi_dot with four or more arguments uses the dynamic programing
+        # optimization and therefore deserve a separate
+        A = np.random.random((6, 2))
```

Comment:
comment seems cut off: "separate test"

These test name are quite long. Rather than `test_basic_function_with` can we have `test_multidot_`, and drop the `_argument`, so this would become `test_multidot_with_dynamic_programming_and_out`.

Reply:
I will commit your suggestion, but also removed the `multidot_with` part, this is already in the `TestMultidot` class, so that should be implicit.

I will then merge this later today or tomorrow.